src/Locator/GuessedPathsLocator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Locator;
+
+use GrumPHP\Configuration\GuessedPaths;
+use GrumPHP\Util\ComposerFile;
+use GrumPHP\Util\Filesystem;
+
+class GuessedPathsLocator
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var GitDirLocator
+     */
+    private $gitDirLocator;
+
+    public function __construct(Filesystem $filesystem, GitDirLocator $gitDirLocator)
+    {
+        $this->filesystem = $filesystem;
+        $this->gitDirLocator = $gitDirLocator;
+    }
+
+    public function locate(?string $cliConfigFile): GuessedPaths
```

Comment:
@Landerstraeten : moved all path guessing logic to this locator.
Can you go through it to see if it makes sense to you?

This one is constructed early in the cli boot code and will be used to tell GrumPHP where to look for stuff. (git, bins, configs, ...)